### DISCUSSION TITLE
Use method expectJsonSnapshot with matcher from cucumber parameter

### DISCUSSION THREAD

**Discussion by tkminh** [pactumjs/pactum#203] [General]
I have cucumber like this:
```cucumber
Then Response should have a status "201"
    And Response should match JSON Snapshot
    """
    {
      "data": {
        "attributes": {
          "reference": 'like("1")'
        }
      }
    }
    """
```

And in my code:

```js
await spec.name(scenarioName).expectJsonSnapshot(matchers);
```

The response return something like:
"data": {
    "attributes": {
      "reference": "1010655",

But when I run, it shows error like:
function has 0 arguments, should have 1 (if synchronous or returning a promise) or 2 (if accepting a callback)

Please support me with this issue.

**Comment by ASaiAnudeep** [comment]
Hello @tkminh ,

We haven't investigated a way to use matchers inside a cucumber file. The only work around is to define custom steps where you can import the matchers and use them.

```js
const pactum = require('pactum');
const { like } = require('pactum-matchers');

// rest of the steps

Then(/^I expect response to match user snapshot$/, async function (snapshotName) {
  await spec.name(snapshotName).expectJsonSnapshot({
    id: like(1)
  })
});
```

**Comment by tkminh** [comment]
Thanks @ASaiAnudeep , can we put the matcher inside the snapshot json file.

**Reply by ASaiAnudeep** [reply]
I don't think it is possible to include matcher inside a json snapshot file.
